frontend/src/q.ts

```diff
@@ -1,13 +1,29 @@
-import { queryOptions } from "@tanstack/react-query";
-import lunr from "lunr";
-import { api } from "./query";
+import { queryOptions, skipToken } from "@tanstack/react-query";
+import { api } from "@/query";
+import { definitions } from "./api";
 
-export const getSearchIndexQuery = () =>
+export const getSearchQuery = (query: string) =>
   queryOptions({
-    queryKey: ["search-index"],
-    queryFn: async () => {
-      const data = await api(`search.json`).json();
+    queryKey: ["search", query],
+    queryFn:
+      query.length > 0
+        ? async ({ signal }) => {
+            // This is the lazy man's debounce, by waiting 100ms before making the request
+            // and utilizing the signal to cancel the request if the query changes
+            // we get a debounce effect
+            await (() => new Promise((resolve) => setTimeout(resolve, 100)))();
+            if (signal.aborted) {
+              return;
+            }
 
-      return lunr.Index.load(data);
-    },
+            const response = await api(
```

Comment:
We only need to await `response.json`, and not even that since we can return it and let React Query handle it.

Reply:
Okay